contracts/StakeHub.sol

```diff
@@ -236,6 +248,24 @@ contract StakeHub is SystemV2, Initializable, Protectable {
         _receiveFundStatus = _DISABLE;
     }
 
+    /**
+     * @dev Modifier to ensure that only an authorized validator can update NodeIDs.
+     * The caller must either be:
+     * 1. The validator itself (i.e. the provided address)
+     * 2. The designated alternative address recorded in `consensusToOperator` for that validator
+     * 3. The BEP-410 authorized sender (agent) for the validator
+     */
+    modifier onlyAuthorized(address validator) {
+        address sender = _bep410MsgSender();
```

Comment:
The `_bep410MsgSender` will return the agent's operator address or itself. So I think the permission checking should be:

```solidity
require(
            _validatorSet.contains(sender) || 
            consensusToOperator[msg.sender] != address(0),
            "Unauthorized caller"
        );
```

and the param can be removed.

Reply:
But the checking seems wrong, if the `msg.sender` is an attack address, it also pass the `sender == validator` checking.

Reply:
I think we can simplify the permission checking~ like this
```solidity
require(
            _validatorSet.contains(sender) || 
            consensusToOperator[msg.sender] != address(0),
            "Unauthorized caller"
        );
```

Reply:
Emm....I got you.
```solidity
require(
            (_validatorSet.contains(sender) && consensusToOperator[validator]==sender) || // checking operator -> validator relationship, if msg.sneder is operator or agent
            (consensusToOperator[validator] != address(0) && msg.sender == validator), // checking if there exist the validator, if msg.sneder is validator
            "Unauthorized caller"
        );
```